XTS/ssl/xts-over-ssh.sh

```diff
@@ -67,56 +72,60 @@ function verify {
-  cli_command $1 "/subsystem=elytron/server-ssl-context=LocalhostSslContext:add(key-manager=LocalhostKeyManager)"
-  cli_command $1 "/subsystem=undertow/server=default-server/https-listener=https:undefine-attribute(name=security-realm)"
-  cli_command $1 "/subsystem=undertow/server=default-server/https-listener=https:write-attribute(name=ssl-context,value=LocalhostSslContext)"
-
-  cli_command $1 ":reload()"
+  local JBOSS_PATH=$1
+  cli_command $JBOSS_PATH $2 "/subsystem=elytron/key-store=LocalhostKeyStore:add(path=server.keystore,relative-to=jboss.server.config.dir,credential-reference={clear-text=\"$3\"},type=JKS)"
+  cli_command $JBOSS_PATH $2 "/subsystem=elytron/key-manager=LocalhostKeyManager:add(key-store=LocalhostKeyStore,alias-filter=\"$3\",credential-reference={clear-text=\"$3\"})"
+  cli_command $JBOSS_PATH $2 "/subsystem=elytron/server-ssl-context=LocalhostSslContext:add(key-manager=LocalhostKeyManager)"
+  cli_command $JBOSS_PATH $2 "/subsystem=undertow/server=default-server/https-listener=https:undefine-attribute(name=security-realm)"
+  cli_command $JBOSS_PATH $2 "/subsystem=undertow/server=default-server/https-listener=https:write-attribute(name=ssl-context,value=LocalhostSslContext)"
+
+  cli_command $JBOSS_PATH $2 ":reload()"
 }
 
 function waitServerStarted() {
-  local PORT=${1:-$CLI_PORT_BASE}
+  local JBOSS_PATH=$1
+  local PORT=${2:-$CLI_PORT_BASE}
   local TIMEOUT=`timeout_adjust 40 2>/dev/null || echo 40` # default timeout is 40 seconds
   local secs=0
 
   until [ $secs -gt $TIMEOUT ]; do
-    "$JBOSS_BIN"/bin/jboss-cli.sh -c --controller=localhost:$PORT --command=":read-attribute(name=server-state)" | grep -s running
+    "$JBOSS_PATH"/bin/jboss-cli.sh -c --controller=localhost:$PORT --command=":read-attribute(name=server-state)" | grep -s running
     if [ $? = 0 ]; then return 0; fi
     ((secs++))
   done
 
   echo "Timeout ${TIMEOUT}s exceeded when waiting for container at port $PORT"
-  "$JBOSS_BIN"/bin/jboss-cli.sh -c --controller=localhost:$PORT --command=":shutdown"
+  "$JBOSS_PATH"/bin/jboss-cli.sh -c --controller=localhost:$PORT --command=":shutdown"
   exit 2
 }
 
 function prepareServerWorkingDirectories() {
-  local JBOSS_HOME_BASE_DIR="${1}"
+  local JBOSS_ZIP_FILE="${1}"
   local JBOSS_TARGET_DIR="${2}"
-  mkdir -p "${JBOSS_TARGET_DIR}/standalone/configuration"
-  mkdir -p "${JBOSS_TARGET_DIR}/standalone/data"
-  mkdir -p "${JBOSS_TARGET_DIR}/standalone/tmp"
-  mkdir -p "${JBOSS_TARGET_DIR}/standalone/log"
-  mkdir -p "${JBOSS_TARGET_DIR}/standalone/content"
-  cp "$JBOSS_HOME_BASE_DIR"/standalone/configuration/*.properties "${JBOSS_TARGET_DIR}/standalone/configuration"
+  unzip "$JBOSS_ZIP_FILE" -d "${JBOSS_TARGET_DIR}"
+
+  local FILENAME=$(basename $JBOSS_ZIP_FILE)
+  local DIRNAME=$(echo "${FILENAME%.*}")
+  mv ${JBOSS_TARGET_DIR}/${DIRNAME}/* ${JBOSS_TARGET_DIR}/
+  rm -rf ${JBOSS_TARGET_DIR}/${DIRNAME}
```

Comment:
I don't think this has to be `rm -rf`, it could be `rm -r`

Reply:
Actually, I think `-f ` is needed to avoid replying to the message "are you sure you want to delete this file?"

Reply:
I think the folder will be empty due to the line 121 so would not need forcing

Reply:
Sorry, I don't even think it should be `rm -r`, it should just be `rm ${JBOSS_TARGET_DIR}/${DIRNAME}`

Reply:
OK, so I think this should be:
` rm -d ${JBOSS_TARGET_DIR}/${DIRNAME}`

${JBOSS_TARGET_DIR}/${DIRNAME} should be empty due to line 121, if it's not then it's a symptom of some problem

Reply:
Done! Thanks Tom